core/domain/platform_parameter_registry.py

```diff
@@ -0,0 +1,300 @@
+            cls, name, committer_id, commit_message, new_rule_dicts):
+        """Updates the platform parameter with new rules.
+
+        Args:
+            name: str. The name of the platform parameter to update.
+            committer_id: str. ID of the committer.
+            commit_message: str. The commit message.
+            new_rule_dicts: list(dist). A list of dict mappings of all fields
+                of PlatformParameterRule object.
+        """
+        param = cls.get_platform_parameter(name)
+
+        # Create a temporary param instance with new rules for validation,
+        # if the new rules are invalid, an exception will be raised in
+        # validate() method.
+        param_dict = param.to_dict()
+        param_dict['rules'] = new_rule_dicts
+        updated_param = param.from_dict(param_dict)
+        updated_param.validate()
+
+        # Set value in datastore.
+        model_instance = config_models.PlatformParameterModel.get(
+            param.name, strict=False)
+        if model_instance is None:
+            model_instance = config_models.PlatformParameterModel.create(
+                param.name,
+                [rule.to_dict() for rule in param.rules],
+                feconf.CURRENT_PLATFORM_PARAMETER_RULE_SCHEMA_VERSION
+            )
+
+        new_rules = [
+            param_domain.PlatformParameterRule.from_dict(
+                rule_dict, param.rule_schema_version)
+            for rule_dict in new_rule_dicts]
+        param.set_rules(new_rules)
+
+        model_instance.rules = [rule.to_dict() for rule in param.rules]
+        model_instance.commit(
+            committer_id,
+            commit_message,
+            [{
+                'cmd': param_domain.PlatformParameterChange.CMD_EDIT_RULES,
+                'new_rules': new_rule_dicts
+            }]
+        )
+
+        memcache_services.delete(
+            param_domain.PlatformParameter.get_memcache_key(name))
+
+    @classmethod
+    def get_all_platform_parameter_names(cls):
+        """Return a list of all the platform parameter names.
+
+        Returns:
+            list(str). The list of all platform parameter names.
+        """
+        return list(cls.parameter_registry.keys())
+
+    @classmethod
+    def evaluate_all_platform_parameters(cls, context):
+        """Evaluate all platform parameters with the given context.
+
+        Args:
+            context: EvaluationContext. The context for evaluation.
+
+        Returns:
+            dict. The keys are the platform parameter names and the values are
+            results of evaluation of the corresponding parameters.
+        """
+        result_dict = {}
+        for parameter_name in cls.get_all_platform_parameter_names():
+            parameter = cls.get_platform_parameter(parameter_name)
+            result_dict[parameter_name] = parameter.evaluate(context)
+        return result_dict
+
+    @classmethod
+    def create_platform_parameter_from_dict(cls, parameter_dict):
```

Comment:
Maybe: ``init_platform_parameter_from_dict`` since this is also doing registration, not just creation.

Reply:
Done.